pages/agent/v3/elastic_ci_aws/troubleshooting.md

```diff
@@ -35,6 +35,20 @@ Resource shortage can cause this issue. See the Auto Scaling group's Activity lo
 
 To fix this issue, change or add more instance types to the `InstanceTypes` template parameter. If 100% of your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting `OnDemandPercentage` parameter to a value above zero.
 
+## Instances are abruptly terminated
+
+This can happen when using Spot Instances. AWS EC2 sends a notification to a spot instance 2 minutes prior to termination. The agent intercepts that notification and attempts to gracefully shut down. If the instance does not shut down gracefully in that time, it is terminated.
+
+To identify if your agent instance was terminated, you can inspect the `/buildkite/lifecycled` CloudWatch log group for the instance. The example below shows the log line indicating that the instance was sent the spot termination notice.
+
+```
+| 2023-07-31 19:19:23.432 | level=info msg="Received termination notice" instanceId=i-abcd notice=spot | i-abcd | 444793955923:/buildkite/lifecycled |
+```
+
+If all your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting the `OnDemandPercentage` parameter to a value above zero.
+
+For better resilience, you can utilise step retries to automatically retry a job that has failed due to spot instance reclamation. See [Automatic retry attributes](/docs/pipelines/command-step#retry-attributes-automatic-retry-attributes) for more information.
```

Comment:
US spelling so `utilise` -> `utilize` but then both are just more complicated ways of saying `use`.

```suggestion
For better resilience, you can use step retries to automatically retry a job that has failed due to spot instance reclamation. See [Automatic retry attributes](/docs/pipelines/command-step#retry-attributes-automatic-retry-attributes) for more information.
```